test/test_optim.py

```diff
@@ -448,6 +448,44 @@ def fwd_bwd(optim, w, b, i):
                 optimizer_c.state_dict()["param_groups"][-1]
             )
 
+    @optims(optim_db, dtypes=[torch.float32])
+    def test_bc_for_state_dict(self, device, dtype, optim_info):
```

Comment:
Should we run part of this as part of https://github.com/pytorch/pytorch/blob/80c07df659362a95da7cd4f3ec367abfdace38c4/.ci/pytorch/test.sh#L835 ?

We can dump state dicts for all models with the old version and load them with the new one!

Bonus point if you make it such that @mikaylagawarecki can easily add the same thing for all Module via ModuleInfo later :)

Reply:
This feels like a larger project and I agree we most likely want e2e tests for load_state_dict somewhere. I'm not sure this exact `test_forward_backward_compatibility` makes the most sense with regards to optimizers though...

Reply:
Okay, after a discussion offline, this ask has been scoped into a good BE task that I've now encapsulated in https://github.com/pytorch/pytorch/issues/116502.

I agree it isn't hard to do but is different enough from this PR.